File: frontend/src/types/pec.ts
export interface MutuelleInfo {
  nom: string
  reseau: string
  reseau_label: string
  email_pec?: string
  portail_url?: string
  portail_label?: string
  telephone?: string
  procedure_email: boolean
  procedure_portail: boolean
  documents_requis: string[]
  delai_reponse_jours: number
  notes?: string
  couleur: string
}

export type StatutPEC =
  | 'brouillon'
  | 'en_attente'
  | 'accordee'
  | 'accordee_partielle'
  | 'refusee'
  | 'annulee'
  | 'expiree'

export type MethodePEC = 'portail_web' | 'email' | 'telephone' | 'courrier'
export type TypeDemande = 'nouvel_appareillage' | 'renouvellement' | 'sav' | 'accessoires'

export interface DocumentPEC {
  id: string
  pec_id: string
  type_document: string
  nom_fichier: string
  mime_type: string
  taille_octets?: number
  created_at: string
}

export interface PEC {
  id: string
  patient_id: string
  author_id: string
  mutuelle_nom: string
  reseau_tiers_payant?: string
  numero_adherent?: string
  type_demande: TypeDemande
  methode: MethodePEC
  statut: StatutPEC
  classe_lpp?: number
  appareil_od_reference?: string
  appareil_og_reference?: string
  montant_demande_od?: number
  montant_demande_og?: number
  montant_accorde_od?: number
  montant_accorde_og?: number
  base_remboursement?: number
  reference_pec?: string
  reference_mutuelle?: string
  numero_dossier_mutuelle?: string
  email_destinataire?: string
  email_envoye_at?: string
  email_objet?: string
  date_demande?: string
  date_reponse?: string
  date_validite?: string
  motif_refus?: string
  notes?: string
  documents: DocumentPEC[]
  created_at: string
}

export interface PECCreate {
  patient_id: string
  mutuelle_nom: string
  reseau_tiers_payant?: string
  numero_adherent?: string
  type_demande?: TypeDemande
  methode?: MethodePEC
  classe_lpp?: number
  appareil_od_reference?: string
  appareil_og_reference?: string
  montant_demande_od?: number
  montant_demande_og?: number
  base_remboursement?: number
  email_destinataire?: string
  notes?: string
}

export interface PECUpdate {
  statut?: StatutPEC
  reference_mutuelle?: string
  numero_dossier_mutuelle?: string
  montant_accorde_od?: number
  montant_accorde_og?: number
  date_validite?: string
  motif_refus?: string
  notes?: string
}

export const STATUT_LABELS: Record<StatutPEC, string> = {
  brouillon: 'Brouillon',
  en_attente: 'En attente',
  accordee: 'Accordée',
  accordee_partielle: 'Accordée partiellement',
  refusee: 'Refusée',
  annulee: 'Annulée',
  expiree: 'Expirée',
}

export const STATUT_COLORS: Record<StatutPEC, string> = {
  brouillon: 'bg-gray-100 text-gray-700',
  en_attente: 'bg-blue-100 text-blue-700',
  accordee: 'bg-green-100 text-green-700',
  accordee_partielle: 'bg-yellow-100 text-yellow-700',
  refusee: 'bg-red-100 text-red-700',
  annulee: 'bg-gray-100 text-gray-500',
  expiree: 'bg-orange-100 text-orange-700',
}

export const TYPE_DEMANDE_LABELS: Record<TypeDemande, string> = {
  nouvel_appareillage: 'Nouvel appareillage',
  renouvellement: 'Renouvellement',
  sav: 'SAV',
  accessoires: 'Accessoires',
}

export const DOC_LABELS: Record<string, string> = {
  devis: 'Devis',
  ordonnance: 'Ordonnance',
  carte_mutuelle: 'Carte mutuelle',
  audiogramme: 'Audiogramme',
  autre: 'Autre document',
}

File: backend/app/schemas/pec.py
from pydantic import BaseModel, EmailStr
from typing import Optional, List
from uuid import UUID
from datetime import datetime
from app.models.pec import StatutPEC, MethodePEC, TypeDemande


class DocumentPECCreate(BaseModel):
    type_document: str  # devis | ordonnance | carte_mutuelle | audiogramme | autre
    nom_fichier: str
    mime_type: str = "application/pdf"
    contenu: Optional[str] = None  # base64


class DocumentPECRead(BaseModel):
    id: UUID
    pec_id: UUID
    type_document: str
    nom_fichier: str
    mime_type: str
    taille_octets: Optional[int]
    created_at: datetime

    model_config = {"from_attributes": True}


class PECCreate(BaseModel):
    patient_id: UUID
    mutuelle_nom: str
    reseau_tiers_payant: Optional[str] = None
    numero_adherent: Optional[str] = None
    type_demande: TypeDemande = TypeDemande.NOUVEL_APPAREILLAGE
    methode: MethodePEC = MethodePEC.EMAIL
    classe_lpp: Optional[int] = 1
    appareil_od_reference: Optional[str] = None
    appareil_og_reference: Optional[str] = None
    montant_demande_od: Optional[float] = None
    montant_demande_og: Optional[float] = None
    base_remboursement: Optional[float] = None
    email_destinataire: Optional[str] = None  # override email auto-détecté
    notes: Optional[str] = None


class PECSendEmailRequest(BaseModel):
    pec_id: UUID
    email_destinataire: Optional[str] = None  # override si besoin
    email_cc: Optional[str] = None
    documents: List[DocumentPECCreate] = []


class PECUpdate(BaseModel):
    statut: Optional[StatutPEC] = None
    reference_mutuelle: Optional[str] = None
    numero_dossier_mutuelle: Optional[str] = None
    montant_accorde_od: Optional[float] = None
    montant_accorde_og: Optional[float] = None
    date_validite: Optional[datetime] = None
    motif_refus: Optional[str] = None
    notes: Optional[str] = None


class PECRead(BaseModel):
    id: UUID
    patient_id: UUID
    author_id: UUID
    mutuelle_nom: str
    reseau_tiers_payant: Optional[str]
    numero_adherent: Optional[str]
    type_demande: TypeDemande
    methode: MethodePEC
    statut: StatutPEC
    classe_lpp: Optional[int]
    appareil_od_reference: Optional[str]
    appareil_og_reference: Optional[str]
    montant_demande_od: Optional[float]
    montant_demande_og: Optional[float]
    montant_accorde_od: Optional[float]
    montant_accorde_og: Optional[float]
    base_remboursement: Optional[float]
    reference_pec: Optional[str]
    reference_mutuelle: Optional[str]
    numero_dossier_mutuelle: Optional[str]
    email_destinataire: Optional[str]
    email_envoye_at: Optional[datetime]
    email_objet: Optional[str]
    date_demande: Optional[datetime]
    date_reponse: Optional[datetime]
    date_validite: Optional[datetime]
    motif_refus: Optional[str]
    notes: Optional[str]
    documents: List[DocumentPECRead] = []
    created_at: datetime

    model_config = {"from_attributes": True}


class MutuelleInfo(BaseModel):
    nom: str
    reseau: str
    reseau_label: str
    email_pec: Optional[str]
    portail_url: Optional[str]
    portail_label: Optional[str]
    telephone: Optional[str]
    procedure_email: bool
    procedure_portail: bool
    documents_requis: List[str]
    delai_reponse_jours: int
    notes: Optional[str]
    couleur: str
